### PR TITLE
This will be a series of small scale and low risk changes for improving startup

### PR DESCRIPTION
**Pull request by ManfredKarrer**

### CODE REVIEW

File: src/main/java/bisq/statistics/StatisticsMain.java [hunk 1]
```diff
@@ -17,175 +17,52 @@
...
-        OptionParser parser = new OptionParser();
-        parser.allowsUnrecognizedOptions();
-        parser.accepts(AppOptionKeys.USER_DATA_DIR_KEY, description("User data directory", DEFAULT_USER_DATA_DIR))
-                .withRequiredArg();
-        parser.accepts(AppOptionKeys.APP_NAME_KEY, description("Application name", DEFAULT_APP_NAME))
-                .withRequiredArg();
+        if (BisqExecutable.setupInitialOptionParser(args)) {
+            // For some reason the JavaFX launch process results in us losing the thread context class loader: reset it.
+            // In order to work around a bug in JavaFX 8u25 and below, you must include the following code as the first line of your realMain method:
+            Thread.currentThread().setContextClassLoader(StatisticsMain.class.getClassLoader());
```
Comment (blabno) [inline src/main/java/bisq/statistics/StatisticsMain.java]: Do we need this javafx hack still?

**Review by blabno — APPROVED** [review]
Tested ACK

File: src/main/java/bisq/statistics/StatisticsMain.java [hunk 2]
```diff
@@ -17,175 +17,52 @@
...
-                        log.error("\n\n############################################################\n" +
-                                        "We shut down as we are over our memory limit. usedMemoryInMB: {}" +
-                                        "\n############################################################\n\n",
-                                finalUsedMemoryInMB);
-                        System.exit(EXIT_FAILURE);
-                    }
-                }, 5);
-            }
-        }, CHECK_MEMORY_PERIOD_SEC);
+        checkMemory(bisqEnvironment, statistics);
```
Comment (blabno) [inline src/main/java/bisq/statistics/StatisticsMain.java]: `keepRunning()` from superclass will hang forever.

File: src/main/java/bisq/statistics/StatisticsMain.java (hunk 1 above)
Comment (ManfredKarrer) [reply src/main/java/bisq/statistics/StatisticsMain.java]: I removed it from the headless apps. I left it for ui app still as I am not sure if it is still an open bug.

**Pull request merged**